core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -382,7 +387,35 @@ public CompletableFuture<Void> maybeInitialize() {
                 }
 
                 // Set the state epoch and end offset from the persisted state.
-                startOffset = partitionData.startOffset() != -1 ? partitionData.startOffset() : 0;
+                if (partitionData.startOffset() != PartitionFactory.DEFAULT_START_OFFSET) {
+                    startOffset = partitionData.startOffset();
+                } else {
+                    GroupConfig.ShareGroupAutoOffsetReset offsetResetStrategy;
+                    if (groupConfigManager.groupConfig(groupId).isPresent()) {
+                        offsetResetStrategy = groupConfigManager.groupConfig(groupId).get().shareAutoOffsetReset();
+                        if (offsetResetStrategy == GroupConfig.ShareGroupAutoOffsetReset.UNKNOWN) {
+                            offsetResetStrategy = GroupConfig.defaultShareAutoOffsetReset();
+                        }
+                    } else {
+                        offsetResetStrategy = GroupConfig.defaultShareAutoOffsetReset();
+                    }
+
+                    if (offsetResetStrategy == GroupConfig.ShareGroupAutoOffsetReset.EARLIEST) {
```

Comment:
I would tend to structure this as a surrounding try-catch, and then the if-else inside. Only needs one catch block, and a bit less duplicated code.

Reply:
Thanks for the review. I have pushed everything into a new method, things should look better after the next commit is pushed.